### ISSUE TITLE
diff from buf from patch fails if any buffer has no newline at e

### ISSUE THREAD

**Issue by albfan** [libgit2/libgit2#5153]
### Reproduction steps

Using the example in #5125 

```
printf 'Hello' > file1
echo Hello > file2
./examples/lg2 diff --no-index file1 file2
diff from patch [-1] - invalid patch hunk at line 7
```
while git works correctly

```
git diff --no-index file1 file2
diff --git 1/file1 2/file2
index 5ab2f8a43..e965047ad 100644
--- 1/file1
+++ 2/file2
@@ -1 +1 @@
-Hello
\ No newline at end of file
+Hello
```

The problem is on https://github.com/libgit2/libgit2/blob/master/src/patch_parse.c#L582 and how to handle lines starting with `\`

### Expected behavior

Output:

```
diff --git 1/file1 2/file2
index 5ab2f8a43..e965047ad 100644
--- 1/file1
+++ 2/file2
@@ -1 +1 @@
-Hello
\ No newline at end of file
+Hello
```

### Actual behavior

```
diff from patch [-1] - invalid patch hunk at line 7
```

### Version of libgit2 (release number or SHA1)

master

### Operating system(s) tested

linux (arch linux)

Related with #5125

**Comment by albfan** [comment]
@pks-t We can dive into this if you want too

```
\ No newline at end of file
```

Are like context lines

**Comment by albfan** [comment]
@neithernut Anyone to ping to fix this? I can implement myself if that would be accepted

**Comment by pks-t** [comment]
@albfan: a fix is already available in #5159

**Comment by ethomson** [comment]
Fixed via #5159
